ImageInLib/ImageInLib/src/procrustrates_dist.c

```diff
@@ -30,23 +31,29 @@ void genProcMeanShape(dataType ** dtaMnShp, Shapes *shapes, size_t height, size_
 		initEstimate[k] = (dataType*)malloc(sizeof(dataType)*(length*width));
 	}
 	//==============================================================================
-	// Choose and initial estimate of the mean shape
-	for (k = 0; k < height; k++)
+	// Struct Pointer to store registered shapes
+	Shapes * reg_shapes = (Shapes*)malloc(numShapes * sizeof(Shapes));
+	dataType initial_value = 0.;
+	for (i = 0; i < numShapes; i++) 
 	{
-		for (i = 0; i < length; i++)
+		reg_shapes[i].num = i + 1;
+		reg_shapes[i].shp = (dataType**)(dataType **)malloc(sizeof(dataType*)*height);
+		for (j = 0; j < height; j++) 
 		{
-			for (j = 0; j < width; j++)
-			{
-				// 1D Conversion of row and column
-				xd = x_new(i, j, length);
-				initEstimate[k][xd] = shapes[params.initEstimateShape].shp[k][xd]; // 1st shape as the estimate initial
-			}
+			reg_shapes[i].shp[j] = (dataType*)malloc(sizeof(dataType)*(length*width));
```

Comment:
also "sizeof(dataType)*(length*width)" shold be extracted as a particular constant